packages/client/src/admin/Apps/Apps.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from 'react';
+import { get } from 'lodash';
+import {
+  Icon,
+  Layout,
+  Card,
+  Row,
+  Col,
+  Button,
+  Input,
+  Typography,
+  Tag,
+  Divider,
+  notification,
+} from 'antd';
+import { graphql } from 'react-apollo';
+import { compose } from 'recompose';
+
+import { errorHandler } from 'utils/errorHandler';
+import Breadcrumbs from 'components/share/breadcrumb';
+import PhAppTemplate from 'interfaces/phAppTemplate';
+import { ActionBtn, ClearBoth, Left } from 'components/apps/detail';
+import { GetPhAppTemplates } from 'queries/PhAppTemplate.graphql';
+import AppLogo from 'components/apps/appLogo';
+
+import { ImportPhAppTemplateFromURL } from './apps.graphql';
+
+const { Search } = Input;
+const { Text } = Typography;
+
+const GITHUB_REGEX =
+  /^https:\/\/github.com\/([A-Za-z0-9._-]+\/[A-Za-z0-9._-]+)\/blob\/(.+)/;
+const GITHUB_REGEX_REPLACE_VALUE = 'https://raw.githubusercontent.com/$1/$2';
+
+interface Props {
+  getPhAppTemplates: any;
+  importPhAppTemplateFromURL: ({
+    variables,
+  }: {
+    variables: { url: string };
+  }) => Promise<void>;
+}
+
+function _Apps({ ...props }: Props) {
+  const importURL = React.useRef(null);
+  const { getPhAppTemplates } = props;
+  const phAppTemplates = get(getPhAppTemplates, 'phAppTemplates', []);
+
+  const [filteredTemplates, setFilteredTemplates] = React.useState([]);
+  const [searchText, setSearchText] = React.useState('');
+
+  const filter = template => {
+    const title = template.name;
+    const description = template.description || '';
+    const index = title.toLowerCase().indexOf(searchText.toLowerCase());
+    const descIndex = description
+      .toLowerCase()
+      .indexOf(searchText.toLowerCase());
+    return index >= 0 || descIndex >= 0;
+  };
+
+  React.useEffect(() => {
+    if (phAppTemplates.length > 0) {
+      setFilteredTemplates(phAppTemplates.filter(filter));
```

Comment:
You don't need to add `filter` in here, because you already handled text change in another effect.

```suggestion
      setFilteredTemplates(phAppTemplates);
```

Reply:
By the way, the `filter` function can be moved into another `useEffect`